### PR TITLE
Add contributing guide

### PR DESCRIPTION
**Pull request by izellevy**
## Solution

Add a guide for contribution.

## Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Infrastructure change (CI configs, etc)
- [x] Non-code change (docs, etc)
- [ ] None of the above: (explain here)

### CODE REVIEW

File: CONTRIBUTING.md
Comment (acatav) [inline CONTRIBUTING.md:1]: can you add a link to this MD from the main MD?

File: CONTRIBUTING.md
Comment (acatav) [inline CONTRIBUTING.md:7]: This is great! I think we should say that also in the main README

**Review by acatav — APPROVED** [review]
LGTM! see my small comments. I suggest to wait for @igiloh-pinecone review before merging.

File: CONTRIBUTING.md
Comment (igiloh-pinecone) [inline CONTRIBUTING.md:82]:
```suggestion
All code changes to Canopy need to be covered by tests. After making your changes, make sure to add the relevant tests.   
If you are adding a new feature or functionality - please add a unit test by creating a new `test_<module_name>.py` file under the `tests/unit/` directory. Tests that require external integration to (e.g. communication with an API or service) should be placed under the `tests/system/` directory. If you are fixing a bug, add a regression test case to the existing test file.   

Please make an effort to avoid code duplication. Some unit tests have a common base class that can be extended. Other tests use fixtures to parameterize the same test cases over several subclasses. Instead of copy-pasting other test cases, try to utilize these mechanism as much as possible.```

File: CONTRIBUTING.md
Comment (igiloh-pinecone) [inline CONTRIBUTING.md:89]: Again, I don't think we should assume `make` as a requirement.  
Either completely replace these with the full command, or at least explain "on linux\mac do this, on windows do that"

File: CONTRIBUTING.md
Comment (igiloh-pinecone) [inline CONTRIBUTING.md:126]:
```suggestion
Once your code has been committed and pushed to Github, all linters and tests are passing - please use Github's website to open a Pull Request.
```

File: CONTRIBUTING.md
Comment (igiloh-pinecone) [inline CONTRIBUTING.md:47]: Don't assume users have `make`.  
Give here the full `poetry install` command.

File: CONTRIBUTING.md
Comment (igiloh-pinecone) [reply CONTRIBUTING.md:1]: +1

There should be a "how to contribute" section in the main MD with a link to this doc

**Review by igiloh-pinecone — COMMENTED** [review]
Hi @izellevy,

I ended up adding quite a few comments. If you prefer we can go over them together.

**Review by igiloh-pinecone — CHANGES_REQUESTED** [review]
See above

File: Makefile
Comment (igiloh-pinecone) [inline Makefile:38]:
```suggestion
	poetry install
```
@izellevy the default install should definitely not be with all (redundant and maybe heavy) extras

**Review by igiloh-pinecone — APPROVED** [review]
LGTM

**Pull request merged**
